### ISSUE TITLE
Suggestion: fake backend for testing

### ISSUE THREAD

**Issue by kpreid** [phaazon/luminance-rs#450]
It is not possible to run code that uses any luminance types (particularly `Tess` and `Texture`) without creating a `GraphicsContext`. It would be convenient for testing if there was a luminance backend that did no actual rendering but provided stub implementations of at least `GraphicsContext`, `Tess`, and `Texture`; this would allow testing logic that is coupled with rendering but does not actually look at pixels.

Even more useful would be if this implementation wasn't purely a no-op but did store data written to `Tess` and `Texture` buffers, so that tests can check what was written to them.

Or, a headless (no visible window) context might also do, but I didn't notice a way to create one — it seems like all paths to get a context go through a windowing interface. (Of course, in WebGL there's always a page anyway, and canvases can be hidden, so that is an option, but it's not an option that works for regular tests running under `cargo test`.)

**Comment by phaazon** [comment]
Agreed, that should be a useful feature and not too complicated. :)
